### PR TITLE
Revert string transcode changes

### PR DESCRIPTION
**Pull request by musm**
add a test and trim leading whitespace

### CODE REVIEW

File: test/mxarray.jl [hunk 1]
```diff
@@ -374,4 +374,5 @@ delete(x)
 
 gc()
 
-
+str = "λ α γ"
+@test jstring(mxarray(str)) == str
```
Comment (yuyichao) [inline test/mxarray.jl]: Missing new line.

**Comment by musm** [comment]
ok fixed. mind merging this one?

File: test/mxarray.jl (hunk 1 above)
Comment (musm) [inline test/mxarray.jl]: why prefer a newline?

**Comment by simonster** [comment]
I'm not sure this is actually right. It seems to round trip, but I think that's just because we aren't properly on either end. If you put UTF-8 into MATLAB you will get UTF-8 out, but MATLAB's internal encoding appears to use UCS-2. Compare:

```julia
julia> str = "λ";

julia> mat"$str(1)+0"
206.0
```

which is UTF-8 vs. entering the same thing into MATLAB directly:

```matlab
>> x = 'λ';
>> x(1)+0

ans =

   955
```

which is the actual Unicode codepoint.

**Comment by musm** [comment]
Interesting, I get the same  on  both on this commit and on master

**Comment by simonster** [comment]
The `transcode` change should have fixed how strings come out of MATLAB, but it didn't fix how we put strings into MATLAB, so that makes sense.

**Comment by simonster** [comment]
See #80 for the fix to encode strings properly on construction.
